src/main/java/org/folio/service/caches/MappingParametersCache.java

```diff
@@ -145,6 +206,115 @@ private List<Future<OrganizationCollection>> getOrganizationCollectionFutures(Ma
+      .map(methodsCollection -> methodsCollection.getAcquisitionMethods()
+        .stream().map(method -> mapTo(method, AcquisitionMethod.class)).toList());
+  }
+
+  private static <T> T mapTo(Object unit, Class<T> objectClass) {
+    return Json.decodeValue(Json.encode(unit), objectClass);
+  }
+
+  private Future<List<String>> getTenantConfigurationAddresses(OkapiConnectionParams params) {
+    return RestUtil.doRequest(params, TENANT_CONFIGURATION_ADDRESSES_URL, HttpMethod.GET, null).compose(httpResponse -> {
+      if ((httpResponse.getResponse().statusCode() == HttpStatus.SC_OK)) {
+        JsonObject response = httpResponse.getJson();
+        if (ifConfigResponseIsValid(response)) {
+          List<String> addresses = response.getJsonArray(CONFIGS_VALUE_RESPONSE).stream()
+            .map(config -> {
+              JsonObject configJson = (JsonObject) config;
+              String configValue = configJson.getString(VALUE_RESPONSE);
+              if (configValue != null) {
+                return new JsonObject(configValue).put(ID, configJson.getString(ID)).toString();
+              }
+              return null;
+            })
+            .filter(Objects::nonNull)
+            .toList();
+          return Future.succeededFuture(addresses);
+        }
+        return Future.succeededFuture(Collections.emptyList());
+      } else {
+        String message = format(ERROR_LOADING_CACHE_MESSAGE, "TenantConfigurationAddresses", params.getTenantId(),
+          httpResponse.getResponse().statusCode(), httpResponse.getBody());
+        LOGGER.warn("getTenantConfigurationAddresses:: {}", message);
+        return Future.failedFuture(new CacheLoadingException(message));
+      }
+    });
+  }
+
+  private <T> Future<List<T>> loadData(OkapiConnectionParams params, String requestUrl, String dataCollectionField,
+                                       Function<JsonObject, List<T>> dataExtractor, String parameterName) {
+    return RestUtil.doRequest(params, requestUrl, HttpMethod.GET, null).compose(httpResponse -> {
+      try {
+        if (httpResponse.getResponse().statusCode() == HttpStatus.SC_OK) {
+          JsonObject response = httpResponse.getJson();
+          if (response != null && response.containsKey(dataCollectionField)) {
+            return Future.succeededFuture(dataExtractor.apply(response));
+          }
+          return Future.succeededFuture(Collections.emptyList());
+        } else {
+          String message = format(ERROR_LOADING_CACHE_MESSAGE, parameterName, params.getTenantId(),
+            httpResponse.getResponse().statusCode(), httpResponse.getBody());
+          LOGGER.warn("loadData:: {}", message);
+          return Future.failedFuture(new CacheLoadingException(message));
+        }
+      } catch (Exception e) {
+        LOGGER.warn("loadData:: Failed to load {}", dataCollectionField, e);
```

Comment:
log.error is more suitable for exceptions

Reply:
according to the folio documentation error level logging should be used for unhandled errors
https://folio-org.atlassian.net/wiki/spaces/DD/pages/1779260/Logging

Reply:
ok. thanks to share